pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -3349,6 +3375,9 @@ type CephBlockPoolRadosNamespaceSpec struct {
 	// the CephBlockPool CR.
 	// +kubebuilder:validation:XValidation:message="blockPoolName is immutable",rule="self == oldSelf"
 	BlockPoolName string `json:"blockPoolName"`
+	// Mirroring configuration of CephBlockPoolRadosNamespaceSpec
+	// +optional
+	Mirroring RadosNamespaceMirroring `json:"mirroring,omitempty"`
```

Comment:
This will allow the mirroring spec to be optional but the spec inside to have required fields.

```suggestion
        // +nullable
	Mirroring *RadosNamespaceMirroring `json:"mirroring,omitempty"`
```

Reply:
Actually, we can still have a namespace created without mirroring as it is optional,
The major question is https://github.com/rook/rook/pull/14701#discussion_r1786115721
` fields are only only mandatory when Enabled is set to true`